lib/pbench/test/unit/server/test_upload.py

```diff
@@ -252,6 +260,46 @@ def test_bad_extension_upload(
         self.verify_logs(caplog)
         assert not self.cachemanager_created
 
+    @pytest.mark.parametrize("error", (errno.ENOSPC, errno.ENFILE, None))
+    def test_bad_stream_read(
+        self, client, server_config, pbench_drb_token, monkeypatch, error
+    ):
+        """Test handling of errors from the intake stream read
+
+        The intake code handles errno.ENOSPC specially; however although the
+        code tried to raise an APIAbort with HTTPStatus.INSUFFICIENT_SPACE
+        (50), the werkzeug abort() doesn't support this and ends up with
+        a generic internal server error. Instead, we now have three distinct
+        cases which all result (to the client) in identical internal server
+        errors. Nevertheless, we exercise all three cases here.
+        """
+        stream = BytesIO(b"12345")
+
+        def access(self, intake: Intake, request: Request) -> Access:
+            return Access(5, stream)
+
+        def read(self):
+            if error:
+                e = OSError()
+                e.errno = error
```

Comment:
Alternately,
```python
                e = OSError(error, "My errno translation")
```
Using [the two+ argument c'tor](https://docs.python.org/3/library/exceptions.html#OSError) produces an experience different from the zero-/one-argument one.  (You can use [`os.strerror`](https://docs.python.org/3/library/os.html#os.strerror`) to generate the second parameter if you like, and you can optionally specify a path-like value for the third parameter, etc.)

Reply:
Yeah, well for some reason my attempts to construct `OSError(errno.ENOSPC)` failed, and I gave up and tried this way. I'm not sure now what was happening, as it works fine in an interactive interpreter today. Weird.

Reply:
You have to use more than one argument to the c'tor, otherwise it falls back to the `Exception()` c'tor (i.e., you get a string-message instead of an errno value).